contracts/token/ERC721.cairo

```diff
@@ -3,165 +3,304 @@
 
 from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
 from starkware.starknet.common.syscalls import get_caller_address
-from starkware.starknet.common.storage import Storage
-from starkware.cairo.common.math import assert_nn_le
+from starkware.cairo.common.math import assert_not_equal, assert_not_zero
+
+# Missing:
+# symbol
+# name
+# tokenURI
+# _baseURI
+
+@storage_var
+func owners(token_id : felt) -> (res : felt):
+end
 
 @storage_var
-func owner(token_id: felt) -> (res: felt):
+func balances(owner : felt) -> (res : felt):
 end
 
 @storage_var
-func balance(owner: felt) -> (res: felt):
+func token_approvals(token_id : felt) -> (res : felt):
 end
 
 @storage_var
-func token_approvals(token_id: felt) -> (res: felt):
+func operator_approvals(owner : felt, operator : felt) -> (res : felt):
 end
 
 @storage_var
-func operator_approvals(owner: felt, operator: felt) -> (res: felt):
+func initialized() -> (res : felt):
+end
+
+@storage_var
+func name_() -> (res : felt):
+end
+
+@storage_var
+func symbol_() -> (res : felt):
+end
+
+struct BlockchainNamespace:
+    member a : felt
+end
+
+# aka ChainID. Chain Agnostic specifies that the length can go up to 32 nines (i.e. 9999999....) but we will only support 31 nines.
+struct BlockchainReference:
+    member a : felt
+end
+
+struct AssetNamespace:
+    member a : felt
+end
+
+# aka contract Address on L1. An address is represented using 20 bytes. Those bytes are written in the `felt`.
+struct AssetReference:
+    member a : felt
+end
+
+# A tokenId is a u256. u256::MAX() is ~1e77 meaning we need 78 characters to store it. Given a felt can represent
```

Comment:
@martriay here's the reasoning behind 3 felts :)
We could use two if we just stored the bytes directly and not the string representation (u256 is 32 bytes, so 1 felt + 1 byte of a second felt)